Gems/Atom/Feature/Common/Assets/Shaders/Materials/BasePBR/BasePBR_LightingData.azsli

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+// This #define magic lets you use the LightingData class in this file without making it the final LightingData
+// used in your shader. Simply #define LightingData to your custom definition before including this file
+// 
+#ifndef LightingData
+#define LightingData        LightingData_BasePBR
+#endif
+
+#include <viewsrg.srgi>
+#include <Atom/Features/Debug.azsli>
+#include <Atom/Features/LightCulling/LightCullingTileIterator.azsli>
+#include <Atom/Features/PBR/LightingUtils.azsli>
+
+class LightingData_BasePBR
```

Comment:
It looks like you are currently using LightingData_BasePBR for all the lighting models. Some of these fields are not used by the BasePBR lighting model, though. For example, emissiveLighting and specularOcclusion are initialized but not used. The compiler should eliminate the dead code, but perhaps we should split out separate LightingData_* structs for the different lighting models with the exact fields that are used for clarity.

Reply:
Will investigate in a follow up PR. I think there were some limitations to omitting certain fields from some structs when I tried to, but can't remember the details.